File: 백준/Bronze/30822. UOSPC 세기/UOSPC 세기.c
#include <stdio.h>
#include <string.h>

int main() {
    int n;
    char S[1001];
    
    scanf("%d", &n);
    getchar();
    fgets(S, sizeof(S), stdin);
    
    int lst[26] = {0};
    for (int i = 0; i < n; i++) {
        if (S[i] >= 'a' && S[i] <= 'z') {
            lst[S[i] - 'a']++;
        }
    }
    

    int dap = 1000;
    char targetChars[] = {'u', 'o', 's', 'p', 'c'};
    for (int i = 0; i < 5; i++) {
        int tmp = lst[targetChars[i] - 'a'];
        if (dap > tmp) {
            dap = tmp;
        }
    }
    printf("%d\n", dap);
    return 0;
}

File: 백준/Bronze/30822. UOSPC 세기/UOSPC 세기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        sc.nextLine();
        String S = sc.nextLine();
        
        int[] lst = new int[26];
        for (int i = 0; i < n; i++) {
            lst[S.charAt(i) - 'a']++;
        }
        
        int dap = 1000;
        char[] targetChars = {'u', 'o', 's', 'p', 'c'};
        for (char ch : targetChars) {
            int tmp = lst[ch - 'a'];
            if (dap > tmp) {
                dap = tmp;
            }
        }
      
        System.out.println(dap);
        sc.close();
    }
}
